syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.exchange";
option java_outer_classname = "ExchangeProto";
option objc_class_prefix = "Xchg";

package exchange;

service ExchangeService {
  rpc ExchangeData(stream ReqChunk) returns (ReplySummary) {}
  rpc ExchangeDataRet(Empty) returns (stream ReqChunk) {}
}

message Empty{
  string name=1;
}

// A helper to stand for void.
message ReplySummary {
  int64 received_chunks=1;
}

// Chunk contains multiple rows data and rows meta.
message ReqChunk {
  bytes data = 1;
  uint64 size = 2;
  uint64 chunk_id=3;
}

